packages/slate-editor/src/extensions/variables/VariableMenu.tsx

```diff
@@ -48,10 +48,6 @@ export function VariableMenu({ container, element, onClose, variables }: Props)
 
     useEffect(() => {
         if (!option?.withFallback) {
-            // Even though the API supports `null` for the fallback,
-            // we can't use it here because Slate will remove this property
-            // and the API will add it back upon saving, creating an endless loop
-            // since the two values will differ.
             updateVariable(editor, { fallback: '' });
         }
     }, [option?.withFallback]);
```

Comment:
Can we not just remove this effect? Why is it necessary? 
Can it be replaced with a normalization function in the VariablesExtension?

Reply:
I believe it is needed because you can change the variable type in the menu and if you change to a type that does not allow a fallback, the previous fallback value (if defined) would stay in the node.

This way, the fallback value will get removed.

Reply:
I believe we sometimes abuse effects for what they are not designed for.

https://react.dev/learn/synchronizing-with-effects#you-might-not-need-an-effect